google/resource_compute_instance_test.go

```diff
@@ -1043,6 +1043,37 @@ func TestAccComputeInstance_secondaryAliasIpRange(t *testing.T) {
 	})
 }
 
+func TestAccComputeInstanceState(t *testing.T) {
+	t.Parallel()
+
+	var instance compute.Instance
+	var instanceName = fmt.Sprintf("instance-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInstance_running(instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeInstanceExists(
+						"google_compute_instance.foobar", &instance),
+					testAccCheckComputeInstanceRunning(&instance),
+				),
+			},
+			{
+				Config: testAccComputeInstance_stopped(instanceName),
```

Comment:
Good start! We'll need to test a handful of cases though.



- [x] You've covered stopping an existing instance
- [x] And deleting a stopped instance
- [ ] Updating a field on a stopped instance
- [ ] Also updating a field that requires stopping, like `machine_type`
- [ ] And doing that at the same time as stopping + starting for each of those
- [ ] Creating a stopped instance

Reply:
Thank you for useful suggestions and review. I've got you point, looks reasonable. I'm going to start with suggested test cases and will ask you for review!